sql-plugin/src/main/spark311/scala/org/apache/spark/sql/rapids/shims/datetimeExpressions.scala

```diff
@@ -107,10 +107,56 @@ case class GpuTimeAdd(
     }
   }
 
+  // A tricky way to check overflow. The result is overflow when positive + positive = negative
+  // or negative + negative = positive, so we can check the sign of the result is the same as
+  // the sign of the operands.
   private def timestampAddDuration(cv: ColumnView, duration: BinaryOperable): ColumnVector = {
     // Not use cv.add(duration), because of it invoke BinaryOperable.implicitConversion,
     // and currently BinaryOperable.implicitConversion return Long
     // Directly specify the return type is TIMESTAMP_MICROSECONDS
-    cv.binaryOp(BinaryOp.ADD, duration, DType.TIMESTAMP_MICROSECONDS)
+    val resWithOverflow = cv.binaryOp(BinaryOp.ADD, duration, DType.TIMESTAMP_MICROSECONDS)
+    closeOnExcept(resWithOverflow) { _ =>
+      val isCvPos = withResource(
```

Comment:
nit: Why not use `AddOverflowChecks.basicOpOverflowCheck`?

https://github.com/NVIDIA/spark-rapids/blob/384b2a95c7fead007136f737624ccb77ec95c0dd/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala#L37

If it does not work because the bit-wise operators don't work timestamp or duration you could use `bitCastTo` along with making a copy of the duration if it is a scalar to make this work.

https://github.com/rapidsai/cudf/blob/72e6f9b08d3c52ca96ed64d963305ab9005ebff6/java/src/main/java/ai/rapids/cudf/ColumnView.java#L2076

We could also refactor it to make the exception code plugable so we can throw what we want/need.

Just a reminder that this is a nit. The code you have looks like it works so you don't have to change it if you don't want to.